src/test/java/com/google/api/codegen/testdata/csharp_gapic_snippets_library.baseline

```diff
@@ -0,0 +1,696 @@
+                }
+            }
+            // End snippet
+        }
+
+        public void ListBooks()
+        {
+            // Snippet: ListBooks(string,string,string,int?,CallSettings)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatShelfName("[SHELF]");
+            string filter = "";
+            // Make the request
+            IPagedEnumerable<ListBooksResponse,Book> response = libraryServiceClient.ListBooks(formattedName, filter);
+
+            // Iterate over all response items, lazily performing RPCs as required
+            foreach (Book item in response)
+            {
+                // Do something with each item
+                Console.Writeline(item);
+            }
+
+            // Or iterate over fixed-sized pages, lazily performing RPCs as required
+            int pageSize = 10;
+            foreach (FixedSizePage<Book> page in response.AsPages().WithFixedSize(pageSize))
+            {
+                // Do something with each page of items
+                Console.Writeline("A page of results:");
+                foreach (Book item in page) {
+                    Console.Writeline(item);
+                }
+            }
+            // End snippet
+        }
+
+        public async Task DeleteBookAsync()
+        {
+            // Snippet: DeleteBookAsync(string,CallSettings)
+            // Additional: DeleteBookAsync(string,CancellationToken)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatBookName("[SHELF]", "[BOOK]");
+            // Make the request
+            await libraryServiceClient.DeleteBookAsync(formattedName);
+            // End snippet
+        }
+
+        public void DeleteBook()
+        {
+            // Snippet: DeleteBook(string,CallSettings)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatBookName("[SHELF]", "[BOOK]");
+            // Make the request
+            libraryServiceClient.DeleteBook(formattedName);
+            // End snippet
+        }
+
+        public async Task UpdateBookAsync()
+        {
+            // Snippet: UpdateBookAsync(string,Book,CallSettings)
+            // Additional: UpdateBookAsync(string,Book,CancellationToken)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatBookName("[SHELF]", "[BOOK]");
+            Book book = new Book();
```

Comment:
It's a shame we don't have a better update story here... maybe this would be a pattern to spot in the future - get the book, modify it, call update...